core/standalone/build.gradle

```diff
@@ -37,30 +38,40 @@ scoverage {
     scoverageScalaVersion.set("${gradle.scala.scoverageScalaVersion}")
 }
 
+task downloadSwagger(type: Download) {
+    src "https://github.com/swagger-api/swagger-ui/archive/v${gradle.swagger.version}.tar.gz"
+    dest file("$buildDir/tmp/swagger/swagger-ui.tar")
+    overwrite false
+}
+
+
 task copySwagger(type: Copy) {
-    def version = "3.6.0"
+    dependsOn(downloadSwagger)
     mkdir("$buildDir/tmp/swagger")
     def destFile = file("$buildDir/tmp/swagger/swagger-ui.tar")
     def uiDir = file("$buildDir/tmp/swagger/swagger-ui")
-    if (!destFile.exists()) {
-        ant.get(src: "https://github.com/swagger-api/swagger-ui/archive/v${version}.tar.gz", dest: destFile)
-    }
     from(tarTree(resources.gzip(destFile))){
-        include("swagger-ui-${version}/dist/**")
+        include("swagger-ui-${gradle.swagger.version}/dist/**")
     }
     into(uiDir)
     includeEmptyDirs = false
-    project.ext.swaggerUiDir = file("$buildDir/tmp/swagger/swagger-ui/swagger-ui-${version}/dist")
+    project.ext.swaggerUiDir = file("$buildDir/tmp/swagger/swagger-ui/swagger-ui-${gradle.swagger.version}/dist")
```

Comment:
just a word of caution:
in general we shouldn't set `project.ext` variables like this inside a task configuration action. if the task is migrated to use the task configuration avoidance API then this variable would not be set.

That being said, I think we should keep it like this for now due to the fact that we don't want to create a massive diff.

Reply:
Just to double-check, do you suggest to remove the introduction of `project.ext` and to use a local variable instead?
That variable would need to be shared between `downloadSwagger` and `copySwagger` tasks.
Extracting the download part from the original task has as goal avoiding to always download the archive.
I'm assume we keep that part, right?